pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/BaseChunkSingleValueReader.java

```diff
@@ -105,17 +140,21 @@ protected ByteBuffer getChunkForRow(int row, ChunkReaderContext context) {
       chunkSize = nextChunkOffset - chunkPosition;
     }
 
-    ByteBuffer uncompressedBuffer = context.getChunkBuffer();
-    uncompressedBuffer.clear();
+    ByteBuffer decompressedBuffer = context.getChunkBuffer();
+    decompressedBuffer.clear();
 
     try {
-      _chunkDecompressor.decompress(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize), uncompressedBuffer);
+      if (_chunkDecompressor != null) {
+        _chunkDecompressor.decompress(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize), decompressedBuffer);
+      } else {
+        decompressedBuffer.put(_dataBuffer.toDirectByteBuffer(chunkPosition, chunkSize));
```

Comment:
why not have no-op implementation for NONE compression case. We can get rid of the if check here. Compiler would optimize it anyway at runtime.

Reply:
Yes, I had thought about that. And convinced myself to not do so because no-op would have required to copy the data, which the initial implementation did not have (line 150). But that led to a bug and I ended up copying the data, so now I can go back to using no-op compressor. Will update.